MSVC 2017 compile error with alignas

I've come across a compile error using msvc2017 and xtensor-io. The error messages aren't terribly informative (to me), but I noticed the usage of `alignas(uint32_t)` inside a for loop, which I thought might be the source msvc's complaint.

https://github.com/xtensor-stack/xtensor-io/blob/392aedaade63e1660728bd2adf694ea2904097d1/include/xtensor-io/xnpz.hpp#L157

The following does make the compiler happy again, but the second answer [here](https://stackoverflow.com/questions/15788947/where-can-i-use-alignas-in-c11) suggests this is incorrect.

```c++
using zip_local_header_aligned = detail::zip_local_header alignas(uint32_t);
for (zip_local_header_aligned entry;;)        
```

If it's alright to bend scoping a bit, this works as well
```c++
alignas(uint32_t) detail::zip_local_header entry;
for (;;)
```

<details>
  <summary>Errors</summary>
  
```
...\xtensor-io\include\xtensor-io\xtensor_io_config.hpp(17): warning C4068: unknown pragma
...\xtensor-io\include\xtensor-io\xtensor_io_config.hpp(18): warning C4068: unknown pragma
...\xtensor-io\include\xtensor-io\xtensor_io_config.hpp(19): warning C4068: unknown pragma
...\xtensor-io\include\xtensor-io/xnpz.hpp(157): error C2061: syntax error: identifier 'zip_local_header'
...\xtensor-io\include\xtensor-io/xnpz.hpp(157): error C2059: syntax error: ')'
...\xtensor-io\include\xtensor-io/xnpz.hpp(159): error C2065: 'entry': undeclared identifier
...\xtensor-io\include\xtensor-io/xnpz.hpp(165): error C2065: 'entry': undeclared identifier
...\xtensor-io\include\xtensor-io/xnpz.hpp(172): error C2065: 'entry': undeclared identifier
...\xtensor-io\include\xtensor-io/xnpz.hpp(173): error C2065: 'entry': undeclared identifier
...\xtensor-io\include\xtensor-io/xnpz.hpp(173): error C2660: 'std::basic_istream<char,std::char_traits<char>>::read': function does not take 1 arguments
C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Tools\MSVC\14.16.27023\include\istream(577): note: see declaration of 'std::basic_istream<char,std::char_traits<char>>::read'
...\xtensor-io\include\xtensor-io/xnpz.hpp(181): error C2065: 'entry': undeclared identifier
...\xtensor-io\include\xtensor-io/xnpz.hpp(185): error C2065: 'entry': undeclared identifier
...\xtensor-io\include\xtensor-io/xnpz.hpp(195): error C2065: 'entry': undeclared identifier
...\xtensor-io\include\xtensor-io/xnpz.hpp(195): error C3861: 'extract_zip64_compressed_size': identifier not found
...\xtensor-io\include\xtensor-io/xnpz.hpp(416): warning C4996: 'localtime': This function or variable may be unsafe. Consider using localtime_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details.
```
</details>

--- a/include/xtensor-io/xnpz.hpp
+++ b/include/xtensor-io/xnpz.hpp
@@ -154,7 +154,8 @@ namespace xt
         using result_type = std::map<std::string, detail::npy_file>;
         result_type arrays;
 
-        for (alignas(uint32_t) detail::zip_local_header entry;;)
+        alignas(uint32_t) detail::zip_local_header entry;
+        for (;;)
         {
             if (!stream.read(reinterpret_cast<char*>(&entry), sizeof(entry)))
             {

